core/src/main/java/com/linecorp/armeria/client/HttpChannelPool.java

```diff
@@ -276,7 +285,13 @@ private boolean usePendingAcquisition(SessionProtocol desiredProtocol, PoolKey k
             if (cause == null) {
                 final SessionProtocol actualProtocol = pch.protocol();
                 if (actualProtocol.isMultiplex()) {
-                    promise.complete(pch);
+                    final HttpSession session = HttpSession.get(pch.get());
+                    final int activeRequests = session.unfinishedResponses() + 1;
+                    if (activeRequests >= session.maxUnfinishedResponses()) {
```

Comment:
Highly unlikely, but `unfinishedResponses + 1` can overflow theoretically. How about this?

    if (session.maxUnfinishedResponses() - session.unfinishedResponses() <= 1) {

or ..

    final int numRemainingStreams =
            session.maxUnfinishedResponses() - session.unfinishedResponses();
    if (numRemainingStreams <= 0) {

Reply:
good point (hadn't really thought about this -- will keep in mind for future PRs as well)  thanks 😄